ios/engine/KMEI/KeymanEngineTests/KMPJSONTests.swift

```diff
@@ -127,4 +129,157 @@ class KMPJSONTests: XCTestCase {
+    guard let constructedMetadata = KMPLexicalModel(from: TestUtils.LexicalModels.mtnt) else {
+      XCTFail("Could not construct KMPLexicalModel from InstallableLexicalModel")
+      return
+    }
+
+    if let _ = KMPLexicalModel(from: TestUtils.Keyboards.khmer_angkor) {
+      XCTFail("Constructed KMPLexicalModel from InstallableKeyboard")
+    }
+
+    // The original version extracted from a KMP.
+    let kmpMetadata: KMPLexicalModel = try loadObjectFromJSON(at: TestUtils.PackageJSON.model_nrc_en_mtnt)!
+
+    XCTAssertEqual(constructedMetadata.id, kmpMetadata.id)
+    XCTAssertEqual(constructedMetadata.name, kmpMetadata.name)
+    // The following line will fail because kmpMetadata.version was only set on the package level,
+    // not directly within the KMPLexicalModel definition.
+    //XCTAssertEqual(constructedMetadata.version, kmpMetadata.version)
+    XCTAssertEqual(constructedMetadata.isRTL, kmpMetadata.isRTL)
+
+    // Language assertions are not as easy.
+    XCTAssertTrue(kmpMetadata.languages.contains(where: { language in
+      // There will only be a single language in our 'constructed' version, since it's built
+      // from a single LanguageResource.  We can't reconstruct the original language name
+      // for lexical models at this time.
+      let constructedLanguage = constructedMetadata.languages[0]
+      return constructedLanguage.languageId == language.languageId
+    }))
+  }
+
+  func testKMPResourceMatches() throws {
+    let kmp_khmer_angkor: KMPKeyboard = try loadObjectFromJSON(at: TestUtils.PackageJSON.keyboard_khmer_angkor)!
+
+    //let wrappedKeyboard = KMPKeyboard(from: TestUtils.Keyboards.khmer_angkor)!
+
+    XCTAssertTrue(kmp_khmer_angkor.matches(installable: TestUtils.Keyboards.khmer_angkor))
+    XCTAssertFalse(kmp_khmer_angkor.matches(installable: TestUtils.LexicalModels.mtnt))
+
+    let kmp_mtnt: KMPLexicalModel = try loadObjectFromJSON(at: TestUtils.PackageJSON.model_nrc_en_mtnt)!
+    kmp_mtnt.setNilVersion(to: "0.1.4")
+
+    XCTAssertTrue(kmp_mtnt.matches(installable: TestUtils.LexicalModels.mtnt))
+    XCTAssertFalse(kmp_mtnt.matches(installable: TestUtils.Keyboards.khmer_angkor))
+
+    // Three models are listed; we choose the second variant since it uses a different language
+    // code than specified under TestUtils.LexicalModels.mtnt.
+    let constructed_mtnt: KMPLexicalModel = KMPLexicalModel(from: kmp_mtnt.installableResources[1])!
+
+    XCTAssertFalse(constructed_mtnt.matches(installable: TestUtils.LexicalModels.mtnt))
+    XCTAssertTrue(constructed_mtnt.matches(installable: TestUtils.LexicalModels.mtnt, requireLanguageMatch: false))
+  }
+
+  func testKMPMetadataFromResource() throws {
+    // https://assertible.com/json-schema-validation#api
+    // - Note:  they "do not make any guarantees about the uptime and availability
+    //          of the Free JSON Schema Validation API."
+    let API_ENDPOINT = URL(string: "https://assertible.com/json")!
```

Comment:
Alternatively, https://github.com/vlas-voloshin/JSONSchemaValidation/ exists, which also suffers the same draft-4 spec limitation as this API, though a bit buggy - it doesn't like our schema's baseline `"#ref"` entry.

There's yet another alternative out there - https://github.com/kylef/JSONSchema.swift - but it's incompatible with Carthage, which is a rather big hurdle to overcome.

Reply:
[For comparison re: Carthage-friendly framework](https://github.com/keymanapp/keyman/compare/feat/ios/engine/kmp-metadata-for-cloud-resource...feat/ios/engine/framework-validation-schema-overhaul)

Ignore the version-only updates.

I developed along this path first, then swapped to use of the online API for validation due to the framework's limitations.  Also, one less Carthage dependency does save build-time, a detail that also factored in to my decision.

Reply:
I'm okay with this as long as we can disable this test in CI.